src/ecdsa/mod.rs

```diff
@@ -257,10 +257,14 @@ impl<C: Signing> Secp256k1<C> {
         &self,
         msg: &Message,
         sk: &SecretKey,
-        noncedata_ptr: *const ffi::types::c_void,
+        noncedata: Option<&[u8; 32]>,
     ) -> Signature {
         unsafe {
             let mut ret = ffi::Signature::new();
+            let noncedata_ptr = match noncedata {
+                Some(arr) => arr.as_c_ptr() as *const _,
+                None => ptr::null(),
+            };
```

Comment:
Oh, you didn't change the signature of `secp256k1_ecdsa_sign` to avoid hassle with `-sys` breaking release?